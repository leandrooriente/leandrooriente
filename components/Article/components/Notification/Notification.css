.notification {
  background: #ffeaa7;
  padding: 20px;
  margin-bottom: 40px;
}