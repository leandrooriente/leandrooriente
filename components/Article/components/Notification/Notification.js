import React from 'react';
import PropTypes from 'prop-types';
import css from './Notification.css';

const Notification = ({ date }) => {
  const articleDate = new Date(date);

  if (Number.isNaN(articleDate.getTime())) {
    return null;
  }

  const actualDate = new Date();
  const articleAge = actualDate.getFullYear() - articleDate.getFullYear();

  if (articleAge <= 2) return null;

  return (
    <div className={css.notification}>
      Este artigo possui mais de {articleAge} anos de existência.
      O seu conteúdo pode estar desatualizado ou pode não representar
      mais minha opinião.
    </div>
  );
};

Notification.propTypes = {
  date: PropTypes.string.isRequired,
};

export default Notification;
