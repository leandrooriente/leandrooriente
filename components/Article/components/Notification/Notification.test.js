import React from 'react';
import { mount, shallow } from 'enzyme';
import Notification from './Notification';

describe('<Notification />', () => {
  const date = new Date();
  const actualYear = date.getFullYear();
  const olderThanTwoYears = `10/10/${actualYear - 3}`;
  const newerThanTwoYears = `10/10/${actualYear - 1}`;

  it('should render <Notification /> component if article is older than 2 years', () => {
    const wrapper = mount(<Notification date={olderThanTwoYears} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should return null if article is newer than 2 years', () => {
    const wrapper = shallow(<Notification date={newerThanTwoYears} />);

    expect(wrapper.getElement()).toBe(null);
  });

  it('should return null if date is invalid', () => {
    const wrapper = shallow(<Notification date="test" />);

    expect(wrapper.getElement()).toBe(null);
  });
});
