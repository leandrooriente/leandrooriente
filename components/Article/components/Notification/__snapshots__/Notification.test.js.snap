// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should render <Notification /> component if article is older than 2 years 1`] = `
<Notification
  date="10/10/2016"
>
  <div
    className="notification"
  >
    Este artigo possui mais de 
    3
     anos de existência. O seu conteúdo pode estar desatualizado ou pode não representar mais minha opinião.
  </div>
</Notification>
`;
